be/src/exprs/map_functions.cpp

```diff
@@ -124,6 +124,76 @@ StatusOr<ColumnPtr> MapFunctions::map_from_arrays(FunctionContext* context, cons
     }
 }
 
+
+StatusOr<ColumnPtr> MapFunctions::map_from_entries(FunctionContext* context, const Columns& columns) {
+    DCHECK_EQ(1, columns.size());
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    auto keys_column = ColumnHelper::unpack_and_duplicate_const_column(columns[0]->size(), columns[0]);
```

Comment:
`unpack_and_duplicate_const_column` will bring additional memory copy and calculation overhead. for  const column, we need to process it separately instead of directly unpacking it

Reply:
for example, you can only process one row and return a const column as result.